src/sudoku.py

```diff
@@ -12,6 +13,21 @@ def __init__(self, grid):
         for i, j in product((0, 3, 6), (0, 3, 6)):
             self.__squares.append([self.__rows[x][y] for x in range(i, i+3) for y in range(j, j+3)])
 
+    def __exclude_equal_alternatives__(self, grid_view):
+        is_any_cell_solved = False  # флаг что хоть одна клетка решена
+        for i in range(9):
+            alternatives = Counter([grid_view[i][j].alternatives for j in range(9)])  # формируем словарь алтернативы в пачке(строка, столбец или квадрант):количество таких альтернатив в пачке 
```

Comment:
1. Нет необходимости оборачивать в список и занимать память, можно передавать генератор
```suggestion
            alternatives = Counter(grid_view[i][j].alternatives for j in range(9))  # формируем словарь альтернативы в пачке(строка, столбец или квадрант):количество таких альтернатив в пачке 
```
2. Не очень удачное именование счетчика. Нужно чтобы было понятно, что это именно счетчик идентичных альтернатив. Например `twin_alternatives_counter`
3. Опечатка в комментарии.